origin-dapp/src/utils/fileUtils.js

```diff
@@ -16,96 +19,69 @@ export const getDataUri = async file => {
  * @param {object} pixelCrop - The object containing the dimensions of the crop area
  * @param {number} pixelCrop.height - The height of the crop area
  * @param {number} pixelCrop.width - The width of the crop area
  * @param {number} pixelCrop.x - The x coordinate of the top-left corner of the crop area
  * @param {number} pixelCrop.y - The y coordinate of the top-left corner of the crop area
- * @param {boolean} skipCropping - If set to true, the image is not cropped, only resized
+ * @param {function} callback- called with the result of modifyImage (an imageDataUri)
  */
-export const generateCroppedImage = async (imageFileObj, pixelCrop, skipCropping = false) => {
-  const MAX_IMAGE_WIDTH = 1000
-  const MAX_IMAGE_HEIGHT = 1000
-  const imageSrc = await getDataUri(imageFileObj)
-  let image
-  let canvas
-  
-  function drawImageOnCanvas(imgEl) {
-    let defaultConfig
-
-    if (skipCropping) {
-      defaultConfig = {
-        x: 0,
-        y: 0,
-        width: imgEl.width,
-        height: imgEl.height
-      }
-    } else {
-      let cropWidth = imgEl.width
-      let cropHeight = imgEl.width / 1.33333 // 4:3 aspect ratio
 
-      if (cropHeight > imgEl.height) {
-        cropHeight = imgEl.height
-        cropWidth = cropHeight * 1.33333
-      }
+export const generateCroppedImage = (imageFileObj, pixelCrop, callback) => {
+  let cropWidth = pixelCrop.width
+  let cropHeight = pixelCrop.width / 1.33333 // 4:3 aspect ratio
 
-      defaultConfig = {
-        x: (imgEl.width - cropWidth) / 2, // center crop horizontally
-        y: (imgEl.height - cropHeight) / 2, // center crop area vertically
-        width: cropWidth,
-        height: cropHeight
-      }
-    }
-
-    const { x, y, width, height } = pixelCrop || defaultConfig
-
-    let resizedWidth = width
-    let resizedHeight = height
-
-    if (width > MAX_IMAGE_WIDTH) {
-      resizedWidth = MAX_IMAGE_WIDTH
-      const widthDiffRatio = resizedWidth / width
-      resizedHeight = height * widthDiffRatio
-    }
-
-    if (resizedHeight > MAX_IMAGE_HEIGHT) {
-      const heightDiffRatio = MAX_IMAGE_HEIGHT / resizedHeight
-      resizedHeight = MAX_IMAGE_HEIGHT
-      resizedWidth = resizedWidth * heightDiffRatio
-    }
+  if (cropHeight > pixelCrop.height) {
+    cropHeight = pixelCrop.height
+    cropWidth = cropHeight * 1.33333
+  }
 
-    canvas = document.createElement('canvas')
-    canvas.width = resizedWidth
-    canvas.height = resizedHeight
-    const ctx = canvas.getContext('2d')
+  const { x, y, width, height } = pixelCrop
 
-    ctx.drawImage(
-      image,
-      x,
-      y,
-      width,
-      height,
-      0,
-      0,
-      resizedWidth,
-      resizedHeight
-    )
+  const options = {
+    maxHeight: MAX_IMAGE_HEIGHT,
+    maxWidth: MAX_IMAGE_WIDTH,
+    minWidth: cropWidth,
+    minHeight: cropHeight,
+    sourceWidth: width,
+    sourceHeight: height,
+    left: x,
+    top: y,
+    orientation: true,
+    crop: true
   }
```

Comment:
ahh ok a couple of other comments on this...
1) When the user is creating a listing, I had it set up to auto-crop the images they select to a 4:3 aspect ratio, taking the full width of the short dimension and the center of the wider dimension. This would be nice to replicate. It currently doesn't look like it's doing any auto-cropping in the listing create workflow.

2) I see now that you're using `top` and `left` to achieve the cropping effect, but I think something fishy about it. It doesn't seem to produce a 4:3 aspect ratio and it doesn't really crop where I expect it to. I suspect something's off with the way `left` and `top` are being set, or maybe `minWidth`, etc?